src/index/IndexImpl.cpp

```diff
@@ -562,9 +563,12 @@ IndexBuilderDataAsStxxlVector IndexImpl::passFileForVocabulary(
       return (*cmp)(a, b, decltype(vocab_)::SortLevel::TOTAL);
     };
     auto wordCallback = vocab_.makeWordWriter(onDiskBase_ + VOCAB_SUFFIX);
+    auto geoWordCallback =
+        vocab_.makeGeoWordWriter(onDiskBase_ + GEO_VOCAB_SUFFIX);
     wordCallback.readableName() = "internal vocabulary";
+    geoWordCallback.readableName() = "internal geometry vocabulary";
     return ad_utility::vocabulary_merger::mergeVocabulary(
-        onDiskBase_, numFiles, sortPred, wordCallback,
+        onDiskBase_, numFiles, sortPred, wordCallback, geoWordCallback,
```

Comment:
As said, the makrWordWriter of the vocabulary should implicitly do the right thing here without any changes required.